kore/src/Kore/Step/Simplification/Ceil.hs

```diff
@@ -313,6 +319,34 @@ makeEvaluateNormalizedAc
             ++ elementsWithVariablesDistinct
     And.simplifyEvaluatedMultiPredicate (MultiAnd.make allConditions)
   where
+    assertCanBeEqCheck
+        :: TermLike variable
+        -> simplifier (TermLike variable)
+    assertCanBeEqCheck key =
+        case key of
+            App_ symbol _ ->
+                if not (Symbol.isInjective symbol)
```

Comment:
`Symbol.isInjective` is redundant because all constructors are injective. Did you mean to check if it's a sort injection?

There is a complication here: This only checks the head of the pattern, but it could have a constructor at the top and unevaluated functions as arguments (for example). I think we may need to follow [this suggestion](https://github.com/kframework/kore/blob/master/kore/src/Kore/Attribute/Symbol.hs#L171) by @virgil-serbanuta and introduce a `NonSimplifiable` pattern attribute to track this.